HTML <blockquote>s are preserved in NSAttributedStrings' NSPresentationIntents

squelart opened the pull request:
#### 2df01fbd82d903cb5db3d42d2cdb4a68b1f542c1
<pre>
HTML &lt;blockquote&gt;s are preserved in NSAttributedStrings&apos; NSPresentationIntents
<a href="https://bugs.webkit.org/show_bug.cgi?id=256514">https://bugs.webkit.org/show_bug.cgi?id=256514</a>
rdar://105853449

Reviewed by Wenson Hsieh.

Block quotes are converted to NSPresentationIntents (of kind NSPresentationIntentKindBlockQuote), which may be nested as appropriate.

* Source/WebCore/PAL/pal/ios/UIKitSoftLink.h:
* Source/WebCore/PAL/pal/ios/UIKitSoftLink.mm:
* Source/WebCore/PAL/pal/spi/cocoa/NSAttributedStringSPI.h:
* Source/WebCore/editing/cocoa/AttributedString.h:
* Source/WebCore/editing/cocoa/AttributedString.mm:
(WebCore::toNSObject):
(WebCore::extractValue):
* Source/WebCore/editing/cocoa/HTMLConverter.mm:
(HTMLConverter::HTMLConverter):
(HTMLConverter::computedAttributesForElement):
(HTMLConverter::_enterBlockquote):
(HTMLConverter::_exitBlockquote):
(HTMLConverter::_enterElement):
(HTMLConverter::_exitElement):
* Source/WebKit/Shared/Cocoa/ArgumentCodersCocoa.mm:
(IPC::shouldEnableStrictMode):
Don&apos;t force strict secure coding on NSPresentationIntents (similar to NSParagraphStyles). Filed rdar://109121874 for follow-up.

* Source/WebKit/Shared/Cocoa/WebCoreArgumentCodersCocoa.serialization.in:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/NSAttributedStringWebKitAdditions.mm:

Canonical link: <a href="https://commits.webkit.org/263887@main">https://commits.webkit.org/263887@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/292d60ee3d9c8e0b478e7bc6870182b68b7ac433

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/6008 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/6185 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/6371 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/7564 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/6365 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/6007 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/6404 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/6142 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/9123 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/6117 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/6153 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/5443 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/7624 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/3629 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/5421 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/13337 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/5495 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/5500 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/7721 "run-api-tests-without-change (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/5953 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/4881 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/5383 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1422 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/9513 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/5747 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

squelart commented on Source/WebKit/Shared/Cocoa/ArgumentCodersCocoa.mm:
```diff
@@ -708,6 +708,10 @@ static bool shouldEnableStrictMode(Decoder& decoder, NSArray<Class> *allowedClas
     if ([allowedClasses containsObject:NSParagraphStyle.class])
         return haveStrictDecodableNSTextTable() && strictSecureDecodingForAllObjCEnabled();
 
+    // rdar://105853449 don't reintroduce rdar://TBC(copy 108038436)
```
Dear reviewer: Once you've allowed this, I'll file the follow-up and put the link here.

whsieh reviewed (approved):
LGTM. Thanks for working on this!

squelart committed:
HTML <blockquote>s are preserved in NSAttributedStrings' NSPresentationIntents
https://bugs.webkit.org/show_bug.cgi?id=256514
rdar://105853449

Reviewed by Wenson Hsieh.

Block quotes are converted to NSPresentationIntents (of kind NSPresentationIntentKindBlockQuote), which may be nested as appropriate.

* Source/WebCore/PAL/pal/ios/UIKitSoftLink.h:
* Source/WebCore/PAL/pal/ios/UIKitSoftLink.mm:
* Source/WebCore/PAL/pal/spi/cocoa/NSAttributedStringSPI.h:
* Source/WebCore/editing/cocoa/AttributedString.h:
* Source/WebCore/editing/cocoa/AttributedString.mm:
(WebCore::toNSObject):
(WebCore::extractValue):
* Source/WebCore/editing/cocoa/HTMLConverter.mm:
(HTMLConverter::HTMLConverter):
(HTMLConverter::computedAttributesForElement):
(HTMLConverter::_enterBlockquote):
(HTMLConverter::_exitBlockquote):
(HTMLConverter::_enterElement):
(HTMLConverter::_exitElement):
* Source/WebKit/Shared/Cocoa/ArgumentCodersCocoa.mm:
(IPC::shouldEnableStrictMode):
Don't force strict secure coding on NSPresentationIntents (similar to NSParagraphStyles). Filed rdar://109121874 for follow-up.

* Source/WebKit/Shared/Cocoa/WebCoreArgumentCodersCocoa.serialization.in:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/NSAttributedStringWebKitAdditions.mm:

Canonical link: https://commits.webkit.org/263887@main
